src/coreComponents/finiteVolume/CellElementStencilTPFA.cpp

```diff
@@ -54,4 +57,23 @@ void CellElementStencilTPFA::add( localIndex const numPts,
   m_connectorIndices[connectorIndex] = oldSize;
 }
 
+void CellElementStencilTPFA::addVectors( real64 const & transMultiplier,
+                                         real64 const (&faceNormal)[3],
+                                         real64 const (&cellToFaceVec)[2][3] )
+{
+  localIndex const oldSize = m_faceNormal.size( 0 );
+  localIndex const newSize = oldSize + 1;
+  m_faceNormal.resize( newSize );
+  m_cellToFaceVec.resize( newSize );
+  m_transMultiplier.resize( newSize );
+
+  m_transMultiplier[oldSize] = transMultiplier;
+
+  LvArray::tensorOps::copy< 3 >( m_faceNormal[oldSize], faceNormal );
+  for( localIndex a=0; a<2; a++ )
+  {
+    LvArray::tensorOps::copy< 3 >( m_cellToFaceVec[oldSize][a], cellToFaceVec[a] );
+  }
```

Comment:
Do you have a mechanism somewhere to `reserve` before filling all these vectors?

Reply:
Unless the stencils were already doing it for all their fields I have not implemented anything specific myself.